pkg/tfbridge/detailed_diff.go

```diff
@@ -264,13 +294,48 @@ func (differ detailedDiffer) makePlainPropDiff(
 	return nil
 }
 
+// makeShortCircuitDiff is used for properties that are nil or computed in either the old or new state.
```

Comment:
This is .. tricky.. If a property is Computed or unknown, we do not know if it will trigger a replacement or not, until we know it. It might trigger a replacement, or it might not. Any implementation here takes a guess and we need to probably call this out in a comment that we are guessing. I would look up current Pulumi behavior for this case and copy how we're guessing to continue in the tradition we have.

Reply:
Added tests in https://github.com/pulumi/pulumi-terraform-bridge/pull/2515. The existing behaviour looks wrong in some cases.